### PR TITLE
refractor!: Implement TODO leftovers

### PR DESCRIPTION
**Pull request by tomfrenken**
This PR is part of the remove deprecated functionality ticket.

Closes SAP/cloud-sdk-backlog#194.

### CODE REVIEW

**Review by FrankEssenberger — COMMENTED** [review]
LGTM @marikaner the legacy code was encapsulated nicely so removal was easy. 👍 I added a minor comment on the exception sentence.

**Review by FrankEssenberger — APPROVED** [review]

File: packages/connectivity/src/scp-cf/authorization-header.ts [hunk 1]
```diff
@@ -184,9 +161,10 @@ function getProxyRelatedAuthHeaders(
     destination.proxyType === 'OnPremise' &&
     destination.authentication === 'NoAuthentication'
   ) {
-    return legacyNoAuthOnPremiseProxy(destination) as any;
+    throw Error(
+      'NoAuthentication is no longer supported for OnPremise connections'
```
Comment (FrankEssenberger) [inline packages/connectivity/src/scp-cf/authorization-header.ts:165]: Perhaps I would repharse this to a more active form: OnPremise connections are not possible with NoAuthentication. Please select a supported auth flow e.g. principal propagation or basic.

Comment (tomfrenken) [reply packages/connectivity/src/scp-cf/authorization-header.ts:165]: True, I'll also add the removed test case back and instead test for the error message

Comment (tomfrenken) [reply packages/connectivity/src/scp-cf/authorization-header.ts:165]: Or is this test case too trivial?

**Pull request merged**